src/main/kotlin/gg/skytils/skytilsmod/utils/RenderUtil.kt

```diff
@@ -758,6 +758,43 @@ object RenderUtil {
         tesselator.draw()
     }
 
+    // see GuiIngame
+    private val vignetteTexPath = ResourceLocation("textures/misc/vignette.png")
+
+    /**
+     * @author Mojang (modified)
+     * @see net.minecraft.client.gui.GuiIngame.renderVignette
+     */
+    fun drawVignette(color: Color) {
+        mc.entityRenderer.setupOverlayRendering()
+        UGraphics.enableBlend()
+        UGraphics.disableDepth()
+
+        UGraphics.depthMask(false)
+        UGraphics.tryBlendFuncSeparate(0, 769, 1, 0)
+
+        // Changing the alpha doesn't affect the vignette, so we have to use the alpha to change the color values
+        UGraphics.color4f(
+            (1f - (color.red / 255f)) * (color.alpha / 255f),
+            (1f - (color.green / 255f)) * (color.alpha / 255f),
+            (1f - (color.blue / 255f)) * (color.alpha / 255f),
+            1f
+        )
+
+        val sr = UResolution
+
+        UGraphics.bindTexture(0, vignetteTexPath)
```

Comment:
To be honest I have no idea what the first parameter (`index`) actually does, but setting it to 0 has not caused any problems

Reply:
I don't know if this method actually exists or not